src/ucp/api/ucp.h

```diff
@@ -707,6 +710,32 @@ typedef struct ucp_worker_params {
      * wakeup.
      */
     unsigned                events;
+
+    /**
+     * User data associated with the current worker.
+     * This value is optional.
+     * If it's not set (along with its corresponding bit in the field_mask -
+     * UCP_WORKER_PARAM_FIELD_USER_DATA), it will default to NULL.
+     */
+    void                    *user_data;
+
+    /**
+     * External event file descriptor.
+     * This value is optional.
+     * If @ref UCP_WORKER_PARAM_FIELD_EVENT_FD is set in the field_mask, events
+     * on the worker will be reported on the provided event file descriptor. In
+     * this case, calling @ref ucp_worker_get_efd() will result in an error.
+     * The provided file descriptor must be capable of aggregating notifications
+     * for arbitrary events, for example @c epoll(7) on Linux systems.
+     * @ref user_data will be used as the event user-data on systems which
+     * support it. For example, on Linux, it will be placed in
+     * @ref epoll_data_t::ptr, when returned from @ref epoll_wait().
```

Comment:
good points. Can we update our hello world code ?

Reply:
this PR updates the implementation and adds a unit test. we can add support for this to hello world in another PR.